src/Options/Generic.hs

```diff
@@ -293,11 +297,23 @@ parseString metavar m =
                    <> Options.long (Data.Text.unpack name)
             Options.option Options.str fs
 
+parseText :: String -> Maybe Text -> Parser Data.Text.Text
+parseText metavar = fmap (fmap Data.Text.pack) (parseString metavar)
+
 instance ParseField Data.Text.Text where
-    parseField = fmap (fmap Data.Text.pack) (parseString "TEXT")
+    parseField = parseText "TEXT"
+
+instance ParseField Data.ByteString.ByteString where
+    parseField = fmap (fmap (Data.Text.Encoding.encodeUtf8)) (parseText "UTF8-BYTESTRING")
```

Comment:
I would still have the field meta-var be "TEXT".  From the command-line user's perspective there is no difference in what they enter.  Only the programmer decoding the result cares that it is encoded into a `ByteString` in a UTF8 format.